tests/integration/generic_test.go

```diff
@@ -28,75 +27,79 @@ var _ = Describe("odo generic", func() {
+			Expect(output).To(ContainSubstring("Component Shortcuts:"))
+			Expect(output).To(ContainSubstring("Flags:"))
+		})
+
+	})
 
-		It("Makes sure that we have the long-description when running odo and we dont error", func() {
+	When("running odo without subcommand and flags", func() {
+		It("a short vesion of help contents is returned, an error is not expected", func() {
 			output := helper.Cmd("odo").ShouldPass().Out()
 			Expect(output).To(ContainSubstring("To see a full list of commands, run 'odo --help'"))
 		})
+	})
 
-		It("Make sure we have the full description when performing odo --help", func() {
-			output := helper.Cmd("odo", "--help").ShouldPass().Out()
-			Expect(output).To(ContainSubstring("Use \"odo [command] --help\" for more information about a command."))
+	When("using an invalid subcommand", func() {
+		It("an error message and help including Usage, Examples, Available commands, and flags is returned", func() {
+			output := helper.Cmd("odo", "hello").ShouldFail().Err()
+			Expect(output).To(ContainSubstring("Usage:"))
+			Expect(output).To(ContainSubstring("Examples:"))
+			Expect(output).To(ContainSubstring("Commands:"))
+			Expect(output).To(ContainSubstring("Flags:"))
 		})
+	})
 
+	When("trying to create a component with an invalid component name", func() {
 		It("Fail when entering an incorrect name for a component", func() {
 			output := helper.Cmd("odo", "component", "foobar").ShouldFail().Err()
-			Expect(output).To(ContainSubstring("Subcommand not found, use one of the available commands"))
+			Expect(output).To(ContainSubstring("Error: Subcommand not found, use one of the available commands:"))
+			Expect(output).To(ContainSubstring("Usage:"))
+			Expect(output).To(ContainSubstring("Examples:"))
+			Expect(output).To(ContainSubstring("Available Commands:"))
+			Expect(output).To(ContainSubstring("Flags:"))
+			Expect(output).To(ContainSubstring("Additional Flags:"))
 		})
-
-		It("Fail with showing help only once for incorrect command", func() {
-			output := helper.Cmd("odo", "hello").ShouldFail().Err()
-			Expect(strings.Count(output, "odo [flags]")).Should(Equal(1))
-		})
-
 	})
 
-	Context("When executing catalog list without component directory", func() {
-		It("should list all component catalogs", func() {
+	When("executing catalog list without component directory", func() {
+		It("should list all devfile components", func() {
 			stdOut := helper.Cmd("odo", "catalog", "list", "components").ShouldPass().Out()
-			helper.MatchAllInOutput(stdOut, []string{"nodejs", "python", "php", "go", "java"})
+			helper.MatchAllInOutput(stdOut, []string{"Devfile", "nodejs", "python", "php", "go", "java"})
 		})
-
 	})
 
-	Context("check catalog component search functionality", func() {
-		It("check that a component does not exist", func() {
+	When("searching the catalog for a non existing component", func() {
+		It("searches for the component and returs message indicating that no component matched the query", func() {
```

Comment:
Same comment as above. Either add a `BeforeEach` block to continue using `When`, or remove it, but ensure you modify your `It` statement accordingly.